docs/dev/release.md

```diff
@@ -1,34 +1,29 @@
+
 ## Create the release branch
 
-Use the GitHub UI to create the new "Release Branch" workflow. Specify the version to be released in the text box.
+Use the GitHub UI to create the new "Create Release Branch" workflow. Specify the version to be released in the text box.
 The deployment scripts (K8s configs, OLM bundle) will be generated and PR will be created with new changes on behalf
 of the `github-actions` bot.
 
-NOTE: The X- and Y- stream releases should only be launched using the workflow from the MAIN branch. Z-stream (patch)
-releases can be launched from a separate branch
-
-The release creation will fail if the file `major-version` contents does not match the major version to be released. This file explicitly means the upcoming release is for a particular major version, with potential breaking changes. This allows us to:
-
-1. Notice if we forgot to update the `major-version` file before releasing the next major version.
-2. Notice if we tried to re-release an older major version when the code is already prepared for the next major version.
-3. Skip some tests, like `helm update`, when crossing from one major version to the next, as such test is not expected to work across incompatible major version upgrades.
+Pass the version with the `X.Y.Z` eg. `1.2.3`, **without** the `v...` prefix.
 
-If the create release branch job fails due an error such as `Bad major version for X... expected Y..`, review whether or not the `major-version` file was updated as expected. Check as well you are not trying to release a patch for the older major version from the new major version codebase.
+See [Troubleshooting](#troubleshooting) in case of issues, such as [errors with the major version](#major-version-issues-when-create-release-branch).
 
 ## Approve the Pull Request named "Release x.y.z"
 
-Review the Pull Request. Approve and merge it to main.
+Review the Pull Request. Approve and merge it to `main`.
 The new job "Create Release" will be triggered and the following will be done:
-* Atlas Operator image built and pushed to dockerhub
+* Atlas Operator image is built and pushed to DockerHub
 * Draft Release will be created with all commits since the previous release
 
-Once the Pull Request is approved, a tag is created out of the branch, which can then be discarded. A branch `release/X.Y.Z` will imply a tag `vX.Y.Z`, and `pre-release/X.Y.Z-...` will imply `vX.Y.Z-...`. The `tag.yml` workflow is the one responsible for creating such tag and triggering the release process workflow (`release-post-merge.yml`).
+Once the Pull Request is approved, a tag is created out of the branch, so such branch could be discarded. A branch `release/X.Y.Z` turns into tag `vX.Y.Z`, and `pre-release/X.Y.Z-...` into `vX.Y.Z-...`. The `tag.yml` workflow is the one responsible for creating a tag from the branch closed, and triggering the release process workflow (`release-post-merge.yml`).
```

Comment:
```suggestion
The "Create Release Branch" workflow is going to create a Pull Request pointing to a `release/X.Y.Z` branch. Once approved and merged, automation is going to create a `vX.Y.Z` tag.
```
I wouldn't mention `tag.yml`. It confuses the person who is doing the release. The statement:
```
The `tag.yml` workflow is the one responsible for creating a tag from the branch closed, and triggering the release process workflow (`release-post-merge.yml`).
```
is rather suited to be placed in the separate markdown which is documenting GitHub actions.